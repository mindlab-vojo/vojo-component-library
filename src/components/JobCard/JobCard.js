import React from 'react'
import PropTypes from 'prop-types'

import * as JobCard from './JobCard.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import { Card } from '../Card/Card.style'
import Typography from '../Typography'

const JobCardComponent = ({
  companyNumber,
  id,
  imageUrl,
  isNew,
  isUnavailable,
  locations,
  onClickCard,
  slots,
  title,
}) => {
  const slotText = slots > 1 ? "VAGAS" : "VAGA"
  const companiesText = companyNumber > 1 ? "empresas" : "empresa"
  
  const renderLocations = (theme) => {
    const locationLength = locations[0].states.length

    switch (locationLength) {
    case 1: 
      return locations[0].states.map(location => (
        <>
          <JobCard.StateCircle 
            key={location.uf}
            theme={theme}>
            <Typography
              fontSize="12px"
              fontWeight="bolder"
            >
              {location.uf}
            </Typography>
          </JobCard.StateCircle>
        </>
      ));
        
    case 27:
      return (
        <JobCard.CountryCircle 
          key={locations[0].country}
          theme={theme}>
          <Typography
            fontSize="12px"
            fontWeight="bolder"
          >
            {locations[0].country}
          </Typography>
        </JobCard.CountryCircle>
      )
      
    default:
      return locations[0].states.map(location => (
        <JobCard.StateCircle 
          key={location.uf}
          theme={theme}>
          <Typography
            fontSize="12px"
            fontWeight="bolder"
          >
            {location.uf}
          </Typography>
        </JobCard.StateCircle>
      ));
    }
  }

  const renderComponent = (theme) => (
    <Card padding="12px 8px">
      <JobCard.Wrapper
        className="JobCard">
        <JobCard.ImageBlock 
          className="JobCard__ImageBlock">
          <JobCard.ImageWrapper
            className="JobCard__ImageWrapper">
            <JobCard.ImageCropper
              className="JobCard__ImageWrapper__Cropper">
              <JobCard.Image
                src={imageUrl}
                alt={title}
              />
            </JobCard.ImageCropper>
          </JobCard.ImageWrapper>
          <JobCard.Slots
            className="JobCard__Slots">
            <Typography 
              fontSize="10px"
              fontWeight="bolder">
              {`${slots} ${slotText}`}
            </Typography>
          </JobCard.Slots>
        </JobCard.ImageBlock>
        <JobCard.DataBlock
          className="JobCard__DataBlock">
          <JobCard.Title
            className="JobCard__Title">
            <Typography
              color={theme.colors.darkBlueColor}
              fontSize="18px"
              fontWeight="bolder">
              {title}
            </Typography>
          </JobCard.Title>
          <JobCard.Companies
            className="JobCard__Companies">
            <Typography
              fontSize="12px">
              {`${companyNumber} ${companiesText} contratando em`}
            </Typography>
          </JobCard.Companies>
          <JobCard.Locations
            className="JobCard__Locations">
            {renderLocations(theme)}
          </JobCard.Locations>
        </JobCard.DataBlock>
      </JobCard.Wrapper>
    </Card>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

JobCardComponent.propTypes = {
  companyNumber: PropTypes.number,
  id: PropTypes.string,
  imageUrl: PropTypes.string,
  isNew: PropTypes.bool,
  isUnavailable: PropTypes.bool,
  locations: PropTypes.array,
  slots: PropTypes.number,
  onClickCard: PropTypes.string,
  title: PropTypes.string,
}

export default JobCardComponent
