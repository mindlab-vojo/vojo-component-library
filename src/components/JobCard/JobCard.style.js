import styled from 'styled-components'

export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  position: relative;
  width: 100%;
`

export const NewTag = styled.div`
  background-color: ${props => props.theme.colors.pinkColor};
  border-radius: 5px;
  padding: 1px 6px 2px;
  position: absolute;
  right: 8px;
  top: -2px;
`

export const ImageBlock = styled.div`
  margin-right: 8px;
  width: 72px;
`

export const ImageWrapper = styled.div``

export const ImageCropper = styled.div`
  display: flex;
  justify-content: center;
  width: 72px;
  height: 72px;
  position: relative;
  overflow: hidden;
  border-radius: ${props => props.borderRadius};
`

export const Image = styled.img`
  display: inline;
  margin: 0 auto;
  height: 100%;
  width: auto;
  z-index: 1;
  ${props => props.isUnavailable && `opacity: 50%;`}
`

export const Slots = styled.div`
  background-color: #FFE010;
  border-radius: 5px;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  margin-top: -18px;
  padding: 5px;
  position: relative;
  width: 72px;
  z-index: 2;
`

export const DataBlock = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: ${props => props.justifyContent};
  min-height: 65px;
`

export const Title = styled.div`
  margin-bottom: 4px;
`

export const Subtitle = styled.div`
  margin-bottom: 4px;
`
export const Companies = styled.div`
  margin: 4px 0;
`
export const Unavailable = styled.div`
  margin: 4px 0;
`

export const Locations = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
`

export const SingleState = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`

export const LocationCircle = styled.div`
  align-items: center;
  background-color: #E9E6FF;
  border-radius: 500px;
  display: flex;
  height: 24px;
  justify-content: center;
  margin: 3px;
  min-width: 24px;
  border: 1px solid #391DDD;
  padding: 2px;
`
export const Freelance = styled.div`
  align-items: center;
  background-color: ${props => props.theme.colors.greyColor};
  border-radius: 500px;
  display: flex;
  height: 24px;
  justify-content: center;
  min-width: 24px;
  padding: 0 10px;
  width: fit-content;
`

export const Cities = styled.div`
  margin-left: 5px;
  width: 100%
`
