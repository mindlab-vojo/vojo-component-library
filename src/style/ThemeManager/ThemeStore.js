import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore