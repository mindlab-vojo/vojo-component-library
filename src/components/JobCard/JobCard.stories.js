import React from 'react';
import { withKnobs, text, number, boolean, object, color } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import JobCard from '.';
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const jobCard = () => {
  const locationsList = [{
    "country": "Brasil",
    "states": ["SP", "RJ"]
  }]
  const borderRadius = text('borderRadius', '50%')
  const companyNumber = number('companyNumber', 5)
  const description = text('description', '')
  const freelance = boolean('freelance', false)
  const id = text('id', 'card-id')
  const imageUrl = text('imageUrl', 'https://cdnb.artstation.com/p/assets/images/images/011/745/187/large/emma-areum-kim-kiki10-01psd.jpg')
  const isNew = boolean('isNew', true)
  const isUnavailable = boolean('isUnavailable', false)
  const justifyContent = text('justifyContent', 'space-between')
  const locations = object('locations', locationsList)
  const locationsOn = boolean('locationsOn', true)
  const onClickCard = action('onClickCard')
  const showCountry = boolean('showCountry', false)
  const slots = number('slots', 200)
  const subtitle = text('subtitle', '')
  const subtitleColor = color('subtitleColor', '')
  const title = text('title', 'Entregador MÃ¡gico')
  const titleSize = text('titleSize', '14px')
  const titleColor = color('titleColor', '')
  const stateNumber = number('stateNumber', 4)

  return (
    <JobCard
      borderRadius={borderRadius}
      companyNumber={companyNumber}
      description={description}
      freelance={freelance}
      id={id}
      imageUrl={imageUrl}
      isNew={isNew}
      isUnavailable={isUnavailable}
      justifyContent={justifyContent}
      locations={locations}
      locationsOn={locationsOn}
      onClickCard={onClickCard}
      showCountry={showCountry}
      slots={slots}
      subtitle={subtitle}
      subtitleColor={subtitleColor}
      title={title}
      titleColor={titleColor}
      titleSize={titleSize}
      stateNumber={stateNumber}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'JobCard',
  component: jobCard,
  decorators: [
    withKnobs,
    withA11y,
    (story) => <ThemeProvider>{story()}</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{story()}</div>,
  ],
}
