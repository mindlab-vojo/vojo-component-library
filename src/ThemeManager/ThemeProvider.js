import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer