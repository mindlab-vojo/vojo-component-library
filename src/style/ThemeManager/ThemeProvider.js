import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer