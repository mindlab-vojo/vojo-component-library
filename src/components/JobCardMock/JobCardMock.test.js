import React from 'react';
import { render } from '@testing-library/react';

import JobCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test JobCard component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <JobCard 
        companyNumber={5}
        id="card-id"
        imageUrl="https://cdnb.artstation.com/p/assets/images/images/011/745/187/large/emma-areum-kim-kiki10-01psd.jpg"
        isNew={true}
        isUnavailable={false}
        locations={[{
          "country": "Brasil",
          "states": ["SP", "RJ"]
        }]}
        onClickCard={fakeFn}
        slots={200}
        title="Entregador MÃ¡gico"/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
