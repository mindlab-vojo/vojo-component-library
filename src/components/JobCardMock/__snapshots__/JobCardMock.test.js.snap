// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test JobCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzokOt hqmVYP"
        id="JobCardMock-card-id"
      >
        <div
          class="sc-AxirZ cLmhxZ"
        >
          <p
            class="sc-fzqBZW kqYVrC"
          >
            novidade
          </p>
        </div>
        <div
          class="sc-AxjAm jDxcNw JobCardMock"
        >
          <div
            class="sc-AxiKw bTpLoE JobCardMock__ImageBlock"
          >
            <div
              class="sc-AxhCb ccMvAG JobCardMock__ImageWrapper"
            >
              <div
                class="sc-AxhUy bZpwYB JobCardMock__ImageWrapper__Cropper"
              >
                <img
                  alt="Entregador M치gico"
                  class="sc-AxgMl kIBdGh"
                  src="https://cdnb.artstation.com/p/assets/images/images/011/745/187/large/emma-areum-kim-kiki10-01psd.jpg"
                />
              </div>
            </div>
            <div
              class="sc-AxheI eKLjGh JobCardMock__Slots"
            >
              <p
                class="sc-fzqNJr cFqHAX"
              >
                RENDA
              </p>
            </div>
          </div>
          <div
            class="sc-Axmtr erLQOu JobCardMock__DataBlock"
          >
            <div
              class="sc-AxmLO cAKCgh JobCardMock__Title"
            >
              <p
                class="sc-fzoyAV infahC"
              >
                Entregador M치gico
              </p>
            </div>
            <div
              class="sc-fzozJi aHfnD JobCardMock__Companies"
            >
              <p
                class="sc-fzoLag fgZswX"
              >
                5 empresas contratando em
              </p>
            </div>
            <div
              class="sc-fzpans iFvVQQ JobCardMock__Locations"
            >
              <div
                class="sc-fznyAO dNloSp"
              >
                <p
                  class="sc-fzoXzr hxNOpN"
                >
                  SP
                </p>
              </div>
              <div
                class="sc-fznyAO dNloSp"
              >
                <p
                  class="sc-fzpjYC dBBYeC"
                >
                  RJ
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzokOt hqmVYP"
      id="JobCardMock-card-id"
    >
      <div
        class="sc-AxirZ cLmhxZ"
      >
        <p
          class="sc-fzqBZW kqYVrC"
        >
          novidade
        </p>
      </div>
      <div
        class="sc-AxjAm jDxcNw JobCardMock"
      >
        <div
          class="sc-AxiKw bTpLoE JobCardMock__ImageBlock"
        >
          <div
            class="sc-AxhCb ccMvAG JobCardMock__ImageWrapper"
          >
            <div
              class="sc-AxhUy bZpwYB JobCardMock__ImageWrapper__Cropper"
            >
              <img
                alt="Entregador M치gico"
                class="sc-AxgMl kIBdGh"
                src="https://cdnb.artstation.com/p/assets/images/images/011/745/187/large/emma-areum-kim-kiki10-01psd.jpg"
              />
            </div>
          </div>
          <div
            class="sc-AxheI eKLjGh JobCardMock__Slots"
          >
            <p
              class="sc-fzqNJr cFqHAX"
            >
              RENDA
            </p>
          </div>
        </div>
        <div
          class="sc-Axmtr erLQOu JobCardMock__DataBlock"
        >
          <div
            class="sc-AxmLO cAKCgh JobCardMock__Title"
          >
            <p
              class="sc-fzoyAV infahC"
            >
              Entregador M치gico
            </p>
          </div>
          <div
            class="sc-fzozJi aHfnD JobCardMock__Companies"
          >
            <p
              class="sc-fzoLag fgZswX"
            >
              5 empresas contratando em
            </p>
          </div>
          <div
            class="sc-fzpans iFvVQQ JobCardMock__Locations"
          >
            <div
              class="sc-fznyAO dNloSp"
            >
              <p
                class="sc-fzoXzr hxNOpN"
              >
                SP
              </p>
            </div>
            <div
              class="sc-fznyAO dNloSp"
            >
              <p
                class="sc-fzpjYC dBBYeC"
              >
                RJ
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
